### PR TITLE
Creates invitation codes

### PR DESCRIPTION
**Pull request by jeffdp**
This change adds the ability for the app to create an invitation code. This allows a remote call to be set up between two instances of the StreemNow sample application.

### CODE REVIEW

**Review by callum-code — APPROVED** [review]
I'm having trouble logging in on my device to test this, but the code looks good

File: StreemNow/ViewController.swift [hunk 1]
```diff
@@ -119,7 +134,20 @@ class ViewController: UIViewControllerSupport {
...
+        }
+    }
+    
+    @IBAction func expertChanged(_ sender: UISwitch) {
+        let expertMode = sender.isOn
+        
+        joinButton.setTitle(expertMode ? "Create" : "Join", for: .normal)
+        invitationTextField.isEnabled = expertMode == false
+        
+        invitationTextField.text = ""
```
[the remark this answers is not in the document]
Comment (dgoldman-pdx) [reply StreemNow/ViewController.swift:150]: This is happening inside an `@IBAction`, so presumably it's already on the main thread.

File: StreemNow/ViewController.swift [hunk 2]
```diff
@@ -119,7 +134,20 @@ class ViewController: UIViewControllerSupport {
...
     @IBAction func joinCallTapped(_ sender: Any) {
-        startCallFromInvite()
+        if expertSwitch.isOn {
+            createInvite()
+        } else {
+            startCallFromInvite()
+        }
+    }
+    
+    @IBAction func expertChanged(_ sender: UISwitch) {
```
[the remark this answers is not in the document]
Comment (dgoldman-pdx) [reply StreemNow/ViewController.swift:144]: Handled at the bottom of this page by `expertSwitch.isEnabled = currentUser.isExpert`?

[the remark this answers is not in the document]
Comment (callum-code) [reply StreemNow/ViewController.swift:144]: ah great!

File: StreemNow/ViewController.swift [hunk 3]
```diff
@@ -165,6 +193,36 @@ class ViewController: UIViewControllerSupport {
         }
         
     }
+    
+    func createInvite() {
+        if let currentUser = StreemInitializer.shared.currentUser {
```
[the remark this answers is not in the document]
Comment (jeffdp) [reply StreemNow/ViewController.swift:198]: Good point! Should be a guard.

File: StreemNow/ViewController.swift [hunk 4]
```diff
@@ -165,6 +193,36 @@ class ViewController: UIViewControllerSupport {
...
+                    if let invitation = invitation {
+                        DispatchQueue.main.async { [weak self] in
+                            guard let self = self else { return }
+                            self.invitationTextField.text = invitation.code.formattedAsInvitationCode()
+                            self.activityIndicator?.stopAnimating()
+                        }
+                    } else {
+                        self.presentAlert(message: "Error creating invitation - \(error.debugDescription)")
+                    }
+                }
```
Comment (dgoldman-pdx) [inline StreemNow/ViewController.swift:222]: Why introduce the new `activityIndicator`? Can't you instead call `self.showActivityIndicator(true)` before calling `createInvitation()`, and then move `self.showActivityIndicator(false)` into both the spot where you now have `self.activityIndicator?.stopAnimating()` and also into the `else` clause?

**Review by dgoldman-pdx — CHANGES_REQUESTED** [review]

File: StreemNow/ViewController.swift (hunk 4 above)
Comment (jeffdp) [reply StreemNow/ViewController.swift:222]: Nice! I didn't realize we already had one available.

Comment (jeffdp) [reply StreemNow/ViewController.swift]: @dgoldman-pdx Done

File: StreemNow/ViewController.swift [hunk 5]
```diff
@@ -165,6 +183,39 @@ class ViewController: UIViewControllerSupport {
...
+        Streem.sharedInstance.createInvitation(forUser: name, referenceId: nil, type: .link) { [weak self] invitation, error in
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                self.showActivityIndicator(false)
+                
+                if let invitation = invitation {
+                    DispatchQueue.main.async { [weak self] in
+                        guard let self = self else { return }
+                        self.invitationTextField.text = invitation.code.formattedAsInvitationCode()
+                        self.showActivityIndicator(false)
```
Comment (dgoldman-pdx) [inline StreemNow/ViewController.swift:211]: Given that you've already called this just above, I guess you don't need to also call it here.

**Review by dgoldman-pdx — APPROVED** [review]
One nit, then 👍

**Pull request merged**